File: feb-06-minimum-deletions-to-make-string-balanced/solution.java
class Solution {
    public int minimumDeletions(String s) {
        int countB = 0;
        int deletions = 0;

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (ch == 'b') {
                countB++;
            } else {
                deletions = Math.min(deletions + 1, countB);
            }
        }

        return deletions;
    }
}

File: feb-06-minimum-deletions-to-make-string-balanced/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        count_b = 0
        deletions = 0

        for ch in s:
            if ch == "b":
                count_b += 1
            else:
                deletions = min(deletions + 1, count_b)

        return deletions
